### PR TITLE
[salus] Salus auth

### PR DESCRIPTION
**Pull request by magx2**
Continuation of: https://github.com/openhab/openhab-addons/pull/16807#discussion_r1623663494

@lsiepel from the doc:

    For now, I would suggest you just copy its content straight into your source code, if there will be any demand, I might publish it on [maven central](https://mvnrepository.com/repos/central).

should I do it?

If yes then should I put it in original package com.frusal.amazonsig4 or in my package?

### CODE REVIEW

File: bundles/org.openhab.binding.salus/src/main/java/com/frusal/amazonsig4/AmazonRequestSignatureV4Utils.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.frusal.amazonsig4;
```
Comment (lsiepel) [inline bundles/org.openhab.binding.salus/src/main/java/com/frusal/amazonsig4/AmazonRequestSignatureV4Utils.java:13]: Package should be adapted

File: bundles/org.openhab.binding.salus/src/main/java/com/frusal/amazonsig4/AmazonRequestSignatureV4Utils.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Copied from https://github.com/frusal/java-amazon-request-signature-v4/blob/master/src/main/java/com/frusal/amazonsig4/AmazonRequestSignatureV4Utils.java
```
[the remark this answers is not in the document]
Comment (lsiepel) [reply bundles/org.openhab.binding.salus/src/main/java/com/frusal/amazonsig4/AmazonRequestSignatureV4Utils.java:25]: Thanks, thanks that is even better!

**Comment by magx2** [comment]
@lsiepel  ping

**Comment by lsiepel** [comment]
> @lsiepel ping

Had some busy days. Will look at this before end of the week.

**Comment by magx2** [comment]
8ts6bp

How it's going @lsiepel ? FYI next 3 weeks I'm on vacation with no access to my IDE

File: bundles/org.openhab.binding.salus/pom.xml [hunk 3]
```diff
@@ -61,5 +59,11 @@
       <version>5.11.0</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.salus/pom.xml:62]: I thought we try to get rid of this dependency? It is in test scope, but why need it there?

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Pull request merged**